src/java/org/apache/cassandra/utils/memory/BufferPool.java

```diff
@@ -920,19 +920,19 @@ private ByteBuffer tryGet(int size, boolean sizeIsLowerBound)
             }
             else if (size > NORMAL_CHUNK_SIZE)
             {
```

Comment:
Overflow is never called?

Reply:
Not sure I follow the context of the comment as this PR intends only to update metrics collected to add micrometer support.

In case you mean why doesn't this appear to allocate off-heap, I think that this `LocalPool.tryGet()` method intentionally does not allocate overflow if the requested size is too big or the pool is exhausted; however `LocalPool.get(int, boolean)` that calls this method will directly call `allocate` which will do so and update `overflowMemoryUsage`.

Reply:
Given https://github.com/riptano/cndb/issues/5691 is going to verify and port DSE metric improvements, maybe this attempt to port just the micrometer changes from https://github.com/riptano/cndb/issues/4974 should just be closed and the micrometer added along with that ticket?